third_party/move/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -275,22 +278,32 @@ pub trait MoveTestAdapter<'a>: Sized {
                     TestRunConfig::CompilerV2 {
                         language_version,
                         v2_experiments,
+                        warnings_are_errors,
                     } => compile_source_unit_v2(
                         state.pre_compiled_deps_v2,
                         state.named_address_mapping.clone(),
                         &state.source_files().cloned().collect::<Vec<_>>(),
                         data_path.to_owned(),
                         self.known_attributes(),
+                        warnings_are_errors,
                         language_version,
                         v2_experiments,
                     )?,
                     // In all other cases, run V1
```

Comment:
In the case of `ComparisonV1V2`, don't we have to run v2 as well?

Reply:
I assume it runs V1 first then calls itself with config=V2 in that case.  Or maybe the `ComparisonV1V2` case never shows up here because it seems to be modified in `run_test_impl` below).  But I don't worry about that, since my change should work ok here, whatever shows up.

Reply:
I've taken your advice below and updated this to be much more like what was originally there.